src/simplifile_erl.erl

```diff
@@ -91,7 +91,9 @@ posix_result(Result) ->
         {ok, Value} ->
             {ok, Value};
         {error, Reason} when ?is_posix_error(Reason) ->
-            {error, Reason}
+            {error, Reason};
+        {error, Reason} ->
+            {error, {unknown, string:uppercase(atom_to_binary(Reason))}}
```

Comment:
The atom_to_binary function defaults to utf8, so it should be fine to pass the resulting binary to string:uppercase